crypto/bio/bss_mem.c

```diff
@@ -311,6 +334,7 @@ static long mem_ctrl(BIO *b, int cmd, long num, void *ptr)
             bm = bbm->buf;
             pptr = (char **)ptr;
             *pptr = (char *)bm;
+            b->flags |= BIO_FLAGS_MEMBUF_FETCH;
```

Comment:
So if someone calls `BIO_C_GET_BUF_MEM_PTR`, we never free the memory ourselves?

We can't do this, it's a significant behaviour change and API break.

An application could absolutely use this to get the `BUF_MEM` without desiring to take ownership of it.